incubator/vault/templates/deployment.yaml

```diff
@@ -26,17 +36,34 @@ spec:
         {{- end }}
         ports:
         - containerPort: {{ .Values.service.port }}
+          name: api
+        - containerPort: 8201
+          name: cluster-address
         livenessProbe:
+          # Alive if it is listening for clustering traffic
           tcpSocket:
             port: {{ .Values.service.port }}
         readinessProbe:
-          tcpSocket:
+          # Ready depends on preference
+          httpGet:
+            path: /v1/sys/health?
+              {{- if .Values.vault.readiness.readyIfSealed -}}sealedcode=204&{{- end }}
+              {{- if .Values.vault.readiness.readyIfStandby -}}standbycode=204&{{- end }}
+              {{- if .Values.vault.readiness.readyIfUninitialized -}}uninitcode=204&{{- end }}
             port: {{ .Values.service.port }}
+            scheme: {{ if .Values.vault.config.listener.tls_disable -}}HTTP{{- else -}}HTTPS{{- end }}
         securityContext:
           readOnlyRootFilesystem: true
           capabilities:
             add:
             - IPC_LOCK
+        env:
```

Comment:
I've also added `VAULT_API_ADDR`  i use here my ingress address.

Reply:
Is that necessary? My understanding is that is only necessary if you're doing client redirection. Since forwarding is default and working that's what I was targeting.

Reply:
I don't remember exactly why I added it, but if i remember correctly it wasn't working without it.
I also use TLS for my cluster communication.

Reply:
Hrm, would you mind trying it as is here? Everything _seems_ to be working for me, but I'm not actually using the ingress either. I'm wondering if maybe there was something up with the ingress controller you were using or perhaps it was just a transient error.

Reply:
@coderanger I am sorry but i don't think i will have time to test as it is.